### ISSUE TITLE
CheckM - lineage_wf fails to complete

### ISSUE THREAD

**Issue by tenguzame** [Ecogenomics/CheckM#108]
Greetings,
I've been using CheckM for quite a long time for my work, but yesterday, all of a sudden, it stopped working.
I had set up a lineage_wf workflow to assess the completeness of some genomic bins I'm currently working on. The program stops at the "analyze" stage with the following error:

Identifying marker genes in 16 bins with 6 threads:
[Error] Failed to set a selected lineage-specific marker set.

What should I do to fix it? I'm on a linux CentOS 7 machine equipped with Python2.7 and Python3.3 on a Miniconda installation.

Michael Tangherlini

**Comment by dparks1134** [comment]
Hello Michael,

Is the problem specific to the 16 bins you are currently processing or does this issue now occur for all bins? Assuming the issue is specific to the 16 bins, can you send these to me so I can investigate the issue?

Cheers,
Donovan

**Comment by lucaz88** [comment]
Dear Donovan,
I am experiencing exactly the same problem. I am analyzing a set of ~400 bacterial genomes and got the same error when it comes to "Identifying marker genes in bins".
I run the same analysis in subsets of the whole dataset to see if the problem was due to some specific fasta file but in all these tests the workflow ended properly. Could it be related with the excessive diversity of the input genomes that hamper the selection of a consistent set of marker genes?

Thanks a lot for helping,
Luca Zoccarato

**Comment by tenguzame** [comment]
Dear all,
I did some further tests with a higher or a lower number of bins to analyze, and it seems that the amount of data to process hinders the capability of the software to analyze genomic bin data. For instance, CheckM seem to perform perfectly with either 1 or 10 bins, but fails to analyze 61 bins. The bins I'm working on are related to a very low number of species, so I don't think it's a matter of genomic diversity (since I recently applied the software on a relatively of high number of bins with a high degree of genomic diversity for a paper we recently published), but maybe the bug might be due to the sheer amount of contigs to analyze.
What do you think?

Michael Tangherlini

**Comment by lucaz88** [comment]
I sure that all my genomes are prokaryotic (although I am running checkM to have an estimation of their quality). Plus I have just run successfully few tests with small subsets of genomes (~15) among which there were pathogens with very streamlined genomes and so probably only few prokaryotic marker genes.

I also succesfully run the lineage_wf using 250 genomes but the problem appears when I use the whole dataset. Accordingly with what you said, it seems to be related with the amount of contigs to analyze however this threshold seems to be somehow related with the input data..
How do you suggest to overcame the problem then? Split the input in several runs and recombine the outputs?

Cheers,
Luca

**Comment by dparks1134** [comment]
Hello. Sorry for the late reply. Michael can you email me the bins that are systematically causing this issue (donovan.parks[at]gmail.com). The number of contigs should not be an issue (I routinely process batches of 1000 genomes), nor should the contents on the bin itself (though, the results on non-prokaryotic bins will not be meaningful).

**Comment by maximilianpress** [comment]
I am experiencing this issue also- did it get resolved ever? i was also able to narrow it down to one bin and the `checkm analyze` step (other steps ran without complaint, generate lineage-specific markers, etc.). I will email this FASTA to Donovan. Thanks.

**Comment by dparks1134** [comment]
Hello Max. Thank you for your message and email. I was able to run your genome through CheckM v1.0.7 without issue:
`checkm lineage_wf -t 16 --pplacer_threads 16 . checkm -x fasta`

The result was:
```
Bin Id                  Marker lineage        # genomes   # markers   # marker sets    0     1    2   3   4   5+ Completeness   Contamination   Strain heterogeneity
trouble_cluster   g__Streptococcus (UID561)       89         679           295        516   158   5   0   0   0       23.69            1.08              20.00
```

The issue seems to be platform dependent as the few other cases reported like this also all ran fine on my system. Is the problem reproducible on your end (i.e., happens when processing only this single bin and happens everytime)?

**Comment by maximilianpress** [comment]
Hi Donovan, thanks for the quick response!

Yes, I was able to narrow it down to this one bin by testing on subsets of
bins, and it happens every time. I will go back and check my versions and
so forth and come back if I find something.

Should have mentioned earlier: I am running this on AWS Ubuntu 14.04 LTS.

Thanks, Max

**Comment by dparks1134** [comment]
Hey Max. Any change you are running lineage_wf with the --reduced_tree flag?

**Comment by maximilianpress** [comment]
hi Donovan- I was in fact using --reduced_tree; sorry I didn't include the
command itself! Might have saved you some time in troubleshooting. I reran
it without this flag and it worked fine (`
checkm lineage_wf -x trouble_cluster.fasta PGScratch/ fail`); I got results
identical to what you show:

`--------------------------------------------------------------------------------------------------------------------------------------------------------------------------

  Bin Id                  Marker lineage        # genomes   # markers   #
marker sets    0     1    2   3   4   5+   Completeness
Contamination   Strain
heterogeneity

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------

  trouble_cluster   g__Streptococcus (UID561)       89         679
  295        516   158   5   0   0   0       23.69            1.08
    20.00

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------`

Whereas when I turn the flag back on it fails again:

`checkm lineage_wf --reduced_tree -x trouble_cluster.fasta PGScratch/
failure

[snip]

*******************************************************************************

 [CheckM - analyze] Identifying marker genes in bins.

*******************************************************************************

  Identifying marker genes in 1 bins with 1 threads:

  [Error] Failed to set a selected lineage-specific marker set.

  [Error] Failed to set a selected lineage-specific marker set.

  Controlled exit resulting from an unrecoverable error or warning.`

Hopefully that helps- let me know if there is some fix or way to guard
against this. And thanks again!

Thanks, Max

**Comment by dparks1134** [comment]
Fixed in v1.0.8.

**Comment by faylward** [comment]
FYI I seem to have come across this issue in checkm v1.0.11. Same as described above- lineage_wf works well but hangs on the "analyze" step, resolved when removing "-r" flag.
